BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/BeaconUtil.java

```diff
@@ -39,6 +46,126 @@ public static float getAdvertisingRange(int transmissionPower) {
         }
     }
 
+    /**
+     * Gets the smallest distance from the given beacons to the user using a filter.
+     *
+     * @param beaconList Beacons to evaluate the distance from
+     * @param filter     Filter for getting the distance of the beacons
+     * @return Distance to the closest beacon; Double.MAX_VALUE if no beacon was given
+     */
+    public static double getSmallestDistance(List<? extends Beacon> beaconList, WindowFilter filter) {
```

Comment:
I think this should internally call `getClosestBeacon(...)` und use the distance from the returned beacon.